FaunaDB.Client/Types/RefParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace FaunaDB.Types
+{
+    internal static class RefParser
+    {
+        public static RefV Parse(Value value)
+        {
+            return value.GetOption(Field.At("@ref")).Match(
+                Some: Parse,
+                None: () =>
+                {
+                    var id = value.GetOption(Field.At("id"));
+                    var cls = value.GetOption(Field.At("class"));
+                    var db = value.GetOption(Field.At("database"));
+
+                    return Mk(id, cls, db);
+                }
+            );
+        }
+
+        static RefV Mk(IOption<Value> id, IOption<Value> cls, IOption<Value> db)
+        {
+            var idE = ParseId(id);
+            var classE = Cast<ClassV>(cls);
+            var databaseE = Cast<DatabaseV>(db);
+
+            if (classE == null && databaseE == null)
+                return BuiltIn.FromName(idE);
+
+            if (classE == null)
```

Comment:
This seems a little odd to me. What are the casses where `classE` could be null?

I was expecting a check for each `BuiltIn` and a `new RefV(...)` at the end if none applies...